libs/blocks/notification/notification.js

```diff
@@ -0,0 +1,130 @@
+  </g>
+  <defs>
+    <clipPath id="clip0_699_20329">
+      <rect width="20" height="20" fill="white"/>
+    </clipPath>
+  </defs>
+</svg>`;
+
+function getOpts(el) {
+  const optRows = [...el.querySelectorAll(':scope > div:nth-of-type(n+3)')];
+  if (!optRows.length) return {};
+  optRows.forEach((row) => row.remove());
+  const camel = (str) => str.replace(/-([a-z])/g, (_, char) => char.toUpperCase());
+  const fmt = (child) => child.textContent.toLowerCase().replace('\n', '').trim();
+  return optRows.reduce((a, c) => ({ ...a, [camel(fmt(c.children[0]))]: fmt(c.children[1]) }), {});
+}
+
+function getBlockData(el) {
+  const variant = variants.find((varClass) => el.classList.contains(varClass)) || defaultVariant;
+  const size = sizes.find((sizeClass) => el.classList.contains(sizeClass)) || defaultSize;
+  const fontSizes = [...blockConfig[variant][size]];
+  if (el.classList.contains('s-button')) fontSizes.splice(3, 1, 'm');
+  return { fontSizes, options: { ...getOpts(el) } };
+}
+
+function wrapCopy(foreground) {
+  const text = foreground.querySelector('.text');
+  if (!text) return;
+  const heading = text?.querySelector('h1, h2, h3, h4, h5, h6');
+  const icon = heading?.previousElementSibling;
+  const body = heading?.nextElementSibling?.classList.contains('action-area') ? '' : heading?.nextElementSibling;
+  const copy = createTag('div', { class: 'copy-wrap' }, [heading, body]);
+  text?.insertBefore(copy, icon?.nextSibling || text.children[0]);
+}
+
+function decorateClose(el) {
+  const btn = createTag('button', { 'aria-label': 'close', class: 'close' }, closeSvg);
+  btn.addEventListener('click', () => (el.style.display = 'none'));
+  el.appendChild(btn);
+}
+
+function decorateFlexible(el) {
+  const innards = [
+    el.querySelector('.background'),
+    el.querySelector('.foreground'),
+    el.querySelector('.close'),
+  ];
+  const inner = createTag('div', { class: 'flexible-inner' }, innards);
+  el.appendChild(inner);
+}
+
+function decorateLayout(el) {
+  const [background, ...rest] = el.querySelectorAll(':scope > div');
+  const foreground = rest.pop();
+  if (background) decorateBlockBg(el, background);
+  foreground?.classList.add('foreground', 'container');
+  const text = foreground?.querySelector('h1, h2, h3, h4, h5, h6, p')?.closest('div');
+  text?.classList.add('text');
+  const picture = text?.querySelector('p picture');
+  const iconArea = picture?.closest('p');
+  iconArea?.classList.add('icon-area');
+  const fgMedia = foreground?.querySelector(':scope > div:not(.text) :is(img, video, a[href*=".mp4"])')?.closest('div');
+  const bgMedia = el.querySelector(':scope > div:not(.foreground) :is(img, video, a[href*=".mp4"])')?.closest('div');
```

Comment:
How would we handle video / mp4? Do we have guidelines for mobile (where videos might not need to get loaded)?

Reply:
That's actually all left entirely to auto-blocks and utility functions like `decorateBlockBg`... so video gets loaded like it normally would everywhere else. The Figma file linked in the Jira ticket in the description here has some guidelines on a lot of the authoring use cases, I'll make sure to get them included in the documentation for this.